src/test/java/info/sjd/controller/OrderControllerTest.java

```diff
@@ -0,0 +1,323 @@
+                build();
+        Cart cart = Cart.builder().
+                closed(0).user(user).
+                creationTime(currentTime).
+                build();
+        Item item = Item.builder().
+                name("test_name").
+                code("t_code").
+                price(123).
+                availability(1).
+                build();
+        Order order = Order.builder().
+                item(item).
+                cart(cart).
+                amount(10).
+                build();
+        when(orderService.findById(anyInt())).thenReturn(order);
+        RequestEntity<Order> requestEntity = new RequestEntity<>(order, HttpMethod.GET, new URI("/order/3"));
+        ResponseEntity<Order> responseEntity = testRestTemplate.exchange(requestEntity, Order.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(orderService, times(1)).findById(anyInt());
+    }
+
+    @Test
+    void findAllTest() throws URISyntaxException {
+        User user = User.builder().
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        Cart cart = Cart.builder().
+                closed(0).user(user).
+                creationTime(currentTime).
+                build();
+        Item item = Item.builder().
+                name("test_name").
+                code("t_code").
+                price(123).
+                availability(1).
+                build();
+        Order order = Order.builder().
+                item(item).
+                cart(cart).
+                amount(10).
+                build();
+        when(orderService.findAll()).thenReturn(Collections.singletonList(order));
+        RequestEntity<Order> requestEntity = new RequestEntity<>(order, HttpMethod.GET, new URI("/order"));
+        ResponseEntity<List> responseEntity = testRestTemplate.exchange(requestEntity, List.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(orderService, times(1)).findAll();
+    }
+
+    @Test
+    void deleteTest() throws URISyntaxException {
+        User user = User.builder().
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        Cart cart = Cart.builder().
+                closed(0).user(user).
+                creationTime(currentTime).
+                build();
+        Item item = Item.builder().
+                name("test_name").
+                code("t_code").
+                price(123).
+                availability(1).
+                build();
+        Order order = Order.builder().
+                item(item).
+                cart(cart).
+                amount(10).
+                build();
+        when(orderService.findById(anyInt())).thenReturn(order);
```

Comment:
New object without fields initialization is enough. It's redundant to initialize other objects.
Also you must mock and check delete method

Reply:
done